core/src/main/scripts/import_data_validator/validateData.py

```diff
@@ -1220,24 +1231,165 @@ class ClinicalValidator(Validator):
     ]
     REQUIRE_COLUMN_ORDER = False
 
+    srv_attrs = None
+
     def __init__(self, *args, **kwargs):
         super(ClinicalValidator, self).__init__(*args, **kwargs)
         self.sampleIds = set()
+        self.attr_defs = []
 
     def processTopLines(self, line_list):
-        """Validate the the attribute definitions above the column header."""
-        # TODO implement this validation
-        pass
+
+        """Parse the the attribute definitions above the column header."""
+
+        LINE_NAMES = ('display_name',
+                      'description',
+                      'datatype',
+                      'attribute_type',
+                      'priority')
+
+        if not line_list:
+            self.logger.error(
+                'No data type header comments found in clinical data file',
+                extra={'line_number': self.line_number})
+            return False
+        if len(line_list) != len(LINE_NAMES):
+            self.logger.error(
+                '%d comment lines at start of clinical data file, expected %d',
+                len(line_list),
+                len(LINE_NAMES))
+            return False
+
+        # remove the # signs
+        line_list = [line[1:] for line in line_list]
+
+        attr_defs = None
+        num_attrs = 0
+        csvreader = csv.reader(line_list,
+                               delimiter='\t',
+                               quoting=csv.QUOTE_NONE,
+                               strict=True)
+        invalid_values = False
+        for line_index, row in enumerate(csvreader):
+
+            if attr_defs is None:
+                # make a list of as many lists as long as there are columns
+                num_attrs = len(row)
+                attr_defs = [OrderedDict() for i in range(num_attrs)]
+            elif len(row) != num_attrs:
+                self.logger.error(
+                    'Varying numbers of columns in clinical header (%d, %d)',
+                    num_attrs,
+                    len(row),
+                    extra={'line_number': line_index + 1})
+                return False
+
+            for col_index, value in enumerate(row):
+
+                # test for invalid values in these (otherwise parseable) lines
+                if value in ('', 'NA'):
+                    self.logger.error(
+                        'Empty %s field in clinical attribute definition',
+                        LINE_NAMES[line_index],
+                        extra={'line_number': line_index + 1,
+                               'column_number': col_index + 1,
+                               'cause': value})
+                    invalid_values = True
+                if LINE_NAMES[line_index] in ('display_name', 'description'):
+                    pass
```

Comment:
It would be, if there were any restrictions on their format other than the ones imposed by the tab-separated file format without a way of escaping tabs and newlines. Whether clinical attribute definitions correspond to the data already in the portal is checked in `ClinicalValidator.checkHeader()`, when parsing the column headers (the IDs of the attributes).